### DISCUSSION TITLE
Revision 3 questions

### DISCUSSION THREAD

**Discussion by MajorPeel** [doudar/SmartSpin2k#433] [Q&A]
Hello everyone,

Firstly, thanks so much to all the contributors on this project, this was exactly what I was looking for to make Zwifting more immersive on my spin bike.

I plan to build Revision 3 of SmartSpin2k in the near future and I have a few questions. 

1. Are links to order the Rev3 PCB live yet? The wiki seems to point to the same PCBWAY link as for Rev 2. eMadmanMakes' Etsy shop seems to only have fully assembled kits available for purchase.
2. I assume that the recommended/common hardware list on the wiki does not apply to Rev3 as lots of the (SMT) components will already be in place on the PCB when it arrives?

I understand that this is a very recent revision of the project and as such, links and instructions may not yet have been updated yet. If the answer is just to be patient, then please just let me know.

Finally, my full time job is as a mechanical design engineer, if there is anything I can contribute to the project on the casing/hardware side the please just let me know.

Thanks

**Comment by doudar** [comment]
Hi @dexter965 ! Yes, lots of links and things that need to be caught up on with the new revision. 

@eMadman currently has some cables and pcb's in stock for revision 3. He should have the full kit listing ready in just a couple days on his etsy shop. If you'd like only a PCB and cable, and not the full kit, message him on etsy or via email and he will work something out with you directly. 

I should have a batch of PCB's and cables shortly as well but it's probably going to be a couple weeks out before mine have a shop listing. All my orders thus far have been small batch test samples but the final ones are in shipping right now.   

As for printing, you'll need:
[Both case halves](https://github.com/doudar/SmartSpin2k/tree/develop/Hardware/V3%20-%20Integrated%20PCB/Case/Case)
[The case window](https://github.com/doudar/SmartSpin2k/blob/develop/Hardware/V3%20-%20Integrated%20PCB/Case/Case/Window.STL)
[Both gears](https://github.com/doudar/SmartSpin2k/tree/develop/Hardware/Common%20Assets/Gears)
[Knob cup](https://github.com/doudar/SmartSpin2k/blob/develop/Hardware/Common%20Assets/KnobCups/Knob_Cup_Large_HEX.STL)
[Knob adaptor (insert) for your bike](https://github.com/doudar/SmartSpin2k/tree/develop/Hardware/Common%20Assets/Inserts)
[Bike Mount](https://github.com/doudar/SmartSpin2k/tree/develop/Hardware/Common%20Assets/Bike%20Mount) 
Arm for the mount [here](https://github.com/doudar/SmartSpin2k/tree/arm_updates/Hardware/Common%20Assets/Arm) or [here](https://github.com/doudar/SmartSpin2k/tree/develop/Hardware/Common%20Assets/Arm)

[and shifters....](https://github.com/doudar/SmartSpin2k/tree/arm_updates/Hardware/Common%20Assets/Shifters)

Speaking of which - If you'd like to do any contributions we could really use some help with shifters. I think we need a total overhaul on how those work. The current design doesn't have the best tactile feedback, can be difficult to assemble, and it's prone to the wires pulling out of the housing if you're not careful handling them. I'd like to see a design that uses a more robust switch in a housing that can be printed in the same materials the case uses. A rocker switch might be a good solution but since we use "holding both buttons" for device reset and for ble scanning, we need to consider that as well. It could be as simple as adding a 2nd reset button to the shifter or something. Additional challenges with a hard plastic case are a bar mount that's strong and can adapt to a wide range of tube diameters, sweat ingress, and reasonably good looks.   

But overall if shifters aren't your thing, we'd love to see contributions anywhere. Thanks for the question!

**Reply by eMadman** [reply]
@dexter965 On the topic of shifters, I have an alternate design here but... it leaves much to be desired: https://github.com/eMadman/SmartSpin2k/tree/plastic-shifter/Hardware/MODs/Plastic%20Shifter

Email me at emadmanmakes@gmail.com and I'll help you get a PCB + wiring harness

**Reply by talan-z** [reply]
Wouldn't these work: https://www.aliexpress.com/item/1005003813401632.html?spm=a2g0o.productlist.main.3.506b2a64WrUXA0&algo_pvid=0e7e98e6-aeda-46f5-9532-830c84fc54d3&algo_exp_id=0e7e98e6-aeda-46f5-9532-830c84fc54d3-1&pdp_ext_f=%7B%22sku_id%22%3A%2212000031154832909%22%7D&pdp_npi=2%40dis%21MYR%2161.48%2143.02%21%21%21%21%21%40211be59e16757284661382624d070e%2112000031154832909%21sea&curPageLogUid=OG9AoSB8k1e7

**Reply by talan-z** [reply]
Hi there,

I just had another thought: There are bluetooth remote controls for steering wheels out there. Why not use one of these (below).
Especially Option 3 and 4 does not look to bad. No hardware changes required.

[Option1: Very small form factor](https://www.aliexpress.com/item/1005004472039795.html?spm=a2g0o.detail.1000060.3.3e3d3b4dFXT7Gf&gps-id=pcDetailBottomMoreThisSeller&scm=1007.13339.291025.0&scm_id=1007.13339.291025.0&scm-url=1007.13339.291025.0&pvid=f38512ff-2513-4cd9-8cf7-c204edd94eec&_t=gps-id:pcDetailBottomMoreThisSeller,scm-url:1007.13339.291025.0,pvid:f38512ff-2513-4cd9-8cf7-c204edd94eec,tpp_buckets:668%232846%238113%231998&pdp_ext_f=%7B%22sku_id%22%3A%2212000029284293796%22%2C%22sceneId%22%3A%223339%22%7D&pdp_npi=2%40dis%21MYR%2124.21%2119.39%21%21%21%21%21%402101f6b216757442471627166ea079%2112000029284293796%21rec)

[Option2: Media buttons](https://www.aliexpress.com/item/32964211450.html?spm=a2g0o.productlist.main.15.285b302f1nLU70&algo_pvid=f419e1db-d94b-4667-ac8a-b6a33709432e&algo_exp_id=f419e1db-d94b-4667-ac8a-b6a33709432e-7&pdp_ext_f=%7B%22sku_id%22%3A%2266495278574%22%7D&pdp_npi=2%40dis%21MYR%2124.04%2117.81%21%21%21%21%21%402102176616757442366563508d068b%2166495278574%21sea&curPageLogUid=T8w8qV5fRB0U)

[Option3: Media control with strap](https://www.aliexpress.com/item/1005004063294911.html?spm=a2g0o.detail.1000060.1.33952564fL4o11&gps-id=pcDetailBottomMoreThisSeller&scm=1007.13339.291025.0&scm_id=1007.13339.291025.0&scm-url=1007.13339.291025.0&pvid=7c030c21-068f-4bbe-b3b6-b225b2681c6c&_t=gps-id:pcDetailBottomMoreThisSeller,scm-url:1007.13339.291025.0,pvid:7c030c21-068f-4bbe-b3b6-b225b2681c6c,tpp_buckets:668%232846%238113%231998&pdp_ext_f=%7B%22sku_id%22%3A%2212000027912105295%22%2C%22sceneId%22%3A%223339%22%7D&pdp_npi=2%40dis%21MYR%2143.86%2132.46%21%21%21%21%21%402101f6b216757443525138930ea079%2112000027912105295%21rec)

[Option4: Media control with strap](https://www.aliexpress.com/item/1005004266606191.html?spm=a2g0o.detail.1000014.7.503a38a0Evd5AY&gps-id=pcDetailBottomMoreOtherSeller&scm=1007.40050.281175.0&scm_id=1007.40050.281175.0&scm-url=1007.40050.281175.0&pvid=1730f872-4771-444f-8625-bf232c367d8e&_t=gps-id:pcDetailBottomMoreOtherSeller,scm-url:1007.40050.281175.0,pvid:1730f872-4771-444f-8625-bf232c367d8e,tpp_buckets:668%232846%238113%231998&pdp_ext_f=%7B%22sku_id%22%3A%2212000028571400371%22%2C%22sceneId%22%3A%2230050%22%7D&pdp_npi=2%40dis%21MYR%2139.17%2127.02%21%21%21%21%21%402101f6b216757443378878645ea079%2112000028571400371%21rec)

**Reply by eMadman** [reply]
Strap styles are generally ideal compared to clamp mount (my opinion, not a fact).  I was looking at pre-manufactured clamp mounts and most were catered towards very specific e-bike and motorcycle handlebar diameters and non-adjustable.  From what I gather, there isn't a standard handlebar diameter for spin bikes so these strap based solutions are more likely to be the best bet - or something that could be easily adapted to strap mounting

I was looking at these sorts of switches as an option since I suspect it wouldn't be too hard to design a handlebar mount for this to attach to
![image](https://user-images.githubusercontent.com/24726844/217264485-0cc9a3d5-051a-4597-8107-5ba54cdf1a8f.png)

**Reply by doudar** [reply]
> Hi there,
> 
> I just had another thought: There are bluetooth remote controls for steering wheels out there. Why not use one of these (below). Especially Option 3 and 4 does not look to bad. No hardware changes required.
> 
> [Option1: Very small form factor](https://www.aliexpress.com/item/1005004472039795.html?spm=a2g0o.detail.1000060.3.3e3d3b4dFXT7Gf&gps-id=pcDetailBottomMoreThisSeller&scm=1007.13339.291025.0&scm_id=1007.13339.291025.0&scm-url=1007.13339.291025.0&pvid=f38512ff-2513-4cd9-8cf7-c204edd94eec&_t=gps-id:pcDetailBottomMoreThisSeller,scm-url:1007.13339.291025.0,pvid:f38512ff-2513-4cd9-8cf7-c204edd94eec,tpp_buckets:668%232846%238113%231998&pdp_ext_f=%7B%22sku_id%22%3A%2212000029284293796%22%2C%22sceneId%22%3A%223339%22%7D&pdp_npi=2%40dis%21MYR%2124.21%2119.39%21%21%21%21%21%402101f6b216757442471627166ea079%2112000029284293796%21rec)
> 
> [Option2: Media buttons](https://www.aliexpress.com/item/32964211450.html?spm=a2g0o.productlist.main.15.285b302f1nLU70&algo_pvid=f419e1db-d94b-4667-ac8a-b6a33709432e&algo_exp_id=f419e1db-d94b-4667-ac8a-b6a33709432e-7&pdp_ext_f=%7B%22sku_id%22%3A%2266495278574%22%7D&pdp_npi=2%40dis%21MYR%2124.04%2117.81%21%21%21%21%21%402102176616757442366563508d068b%2166495278574%21sea&curPageLogUid=T8w8qV5fRB0U)
> 
> [Option3: Media control with strap](https://www.aliexpress.com/item/1005004063294911.html?spm=a2g0o.detail.1000060.1.33952564fL4o11&gps-id=pcDetailBottomMoreThisSeller&scm=1007.13339.291025.0&scm_id=1007.13339.291025.0&scm-url=1007.13339.291025.0&pvid=7c030c21-068f-4bbe-b3b6-b225b2681c6c&_t=gps-id:pcDetailBottomMoreThisSeller,scm-url:1007.13339.291025.0,pvid:7c030c21-068f-4bbe-b3b6-b225b2681c6c,tpp_buckets:668%232846%238113%231998&pdp_ext_f=%7B%22sku_id%22%3A%2212000027912105295%22%2C%22sceneId%22%3A%223339%22%7D&pdp_npi=2%40dis%21MYR%2143.86%2132.46%21%21%21%21%21%402101f6b216757443525138930ea079%2112000027912105295%21rec)
> 
> [Option4: Media control with strap](https://www.aliexpress.com/item/1005004266606191.html?spm=a2g0o.detail.1000014.7.503a38a0Evd5AY&gps-id=pcDetailBottomMoreOtherSeller&scm=1007.40050.281175.0&scm_id=1007.40050.281175.0&scm-url=1007.40050.281175.0&pvid=1730f872-4771-444f-8625-bf232c367d8e&_t=gps-id:pcDetailBottomMoreOtherSeller,scm-url:1007.40050.281175.0,pvid:1730f872-4771-444f-8625-bf232c367d8e,tpp_buckets:668%232846%238113%231998&pdp_ext_f=%7B%22sku_id%22%3A%2212000028571400371%22%2C%22sceneId%22%3A%2230050%22%7D&pdp_npi=2%40dis%21MYR%2139.17%2127.02%21%21%21%21%21%402101f6b216757443378878645ea079%2112000028571400371%21rec)

Unfortunately BLE HID currently doesn’t work with NimBLE (the Bluetooth library we use) as a Client. The great news is that @h2zero is working on this and hopefully shortly we will be able to use BLE HID devices as input. 

My target is [BLE camera shutter buttons for phones](https://a.co/d/aSvAX3e). They are under $10 each and could be placed in a tpu band to make an easy no-solder wireless shifter.

**Reply by talan-z** [reply]
Thanks. The BLE camera shutter buttons would be more or less in line with Option 3 or 4 (Media control with strap). Looks much simpler than the current shifter setup.

I have found the following code snippet [BLE HID client with NimBLE](https://github.com/esp32beans/BLE_HID_Client)
Excuse my ignorance/lack of understanding, but isn't that the BLE HID client setup or am I missing something?

**Reply by doudar** [reply]
> Thanks. The BLE camera shutter buttons would be more or less in line with Option 3 or 4 (Media control with strap). Looks much simpler than the current shifter setup.
> 
> 
> 
> I have found the following code snippet [BLE HID client with NimBLE](https://github.com/esp32beans/BLE_HID_Client)
> 
> Excuse my ignorance/lack of understanding, but isn't that the BLE HID client setup or am I missing something?

Looks like a great find! I tried a similar projects code a few weeks back and couldn't get callbacks working. We need to play with this more. I think it would be a great solution.

**Reply by talan-z** [reply]
Happy to hear that! I ordered now all parts for v2 including some bluetooth HID devices from aliexpress. Will take a month or so until it is here. Once setup, will also play with the HID client thing a bit more.

**Comment by MajorPeel** [comment]
Thanks very much for the quick and detailed answers @doudar @eMadman. I will email directly regarding the PCB and wiring.

Sure I'd love to take a look at the shifters. I'll let you know when I have something worth sharing.

**Comment by doudar** [comment]
> Hello everyone,
> 
> Firstly, thanks so much to all the contributors on this project, this was exactly what I was looking for to make Zwifting more immersive on my spin bike.
> 
> I plan to build Revision 3 of SmartSpin2k in the near future and I have a few questions.
> 
> 1. Are links to order the Rev3 PCB live yet? The wiki seems to point to the same PCBWAY link as for Rev 2. eMadmanMakes' Etsy shop seems to only have fully assembled kits available for purchase.
> 2. I assume that the recommended/common hardware list on the wiki does not apply to Rev3 as lots of the (SMT) components will already be in place on the PCB when it arrives?
> 
> I understand that this is a very recent revision of the project and as such, links and instructions may not yet have been updated yet. If the answer is just to be patient, then please just let me know.
> 
> Finally, my full time job is as a mechanical design engineer, if there is anything I can contribute to the project on the casing/hardware side the please just let me know.
> 
> Thanks

New Rev 3 building instructions!
https://github.com/doudar/SmartSpin2k/blob/develop/SS2kR3BuildingInstructions.pdf
